### PR TITLE
change lib to use laminas instead of zend

### PR DESCRIPTION
**Pull request by habibillah**
- change lib to use laminas instead of zend
- work with slm/queue:^2.0
- work with latest phpunit

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,18 +8,10 @@ cache:
...
 php:
-  - 5.5
-  - 5.6
-  - 7.0
-
-env:
-  matrix:
-    - PREFER_LOWEST="--prefer-lowest"
-    - PREFER_LOWEST=""
+  - 7.2
```
Comment (roelvanduijnhoven) [inline .travis.yml:11]: I would suggest to copy the stuff from https://github.com/JouwWeb/SlmQueue/blob/master/.travis.yml. That way you test more future versions, and the repos are more in sync :).

File: composer.json [hunk 2]
```diff
@@ -24,23 +24,24 @@
...
     ],
     "require": {
-        "php": "^5.5 || ^7.0",
-        "zendframework/zend-eventmanager": "^2.6 || ^3.0",
-        "zendframework/zend-servicemanager": "^2.7 || ^3.0",
-        "zendframework/zend-stdlib": "^2.2 || ^3.0",
-        "slm/queue": "^1.0",
+        "ext-json": "*",
+        "php": "^7.2",
+        "laminas/laminas-eventmanager": "^2.6 || ^3.0",
```
Comment (roelvanduijnhoven) [inline composer.json:29]:
```suggestion
        "laminas/laminas-eventmanager": "^3.2.1",
```

File: composer.json [hunk 3]
```diff
@@ -24,23 +24,24 @@
...
+        "laminas/laminas-servicemanager": "^2.7 || ^3.0",
```
Comment (roelvanduijnhoven) [inline composer.json:30]:
```suggestion
        "laminas/laminas-servicemanager": "^3.3.1",
```

File: composer.json [hunk 4]
```diff
@@ -24,23 +24,24 @@
...
+        "laminas/laminas-stdlib": "^2.2 || ^3.0",
```
Comment (roelvanduijnhoven) [inline composer.json:31]:
```suggestion
        "laminas/laminas-stdlib": "^3.2",
```

File: composer.json [hunk 5]
```diff
@@ -24,23 +24,24 @@
...
+        "slm/queue": "^2.0",
         "aws/aws-sdk-php-zf2": "^3.0 || ^4.0"
```
Comment (roelvanduijnhoven) [inline composer.json:33]: Would it be an idea to bump this to >= 4.0?

File: src/Queue/SqsQueue.php [hunk 6]
```diff
@@ -29,15 +29,15 @@ class SqsQueue extends AbstractQueue implements SqsQueueInterface
     /**
      * Constructor
      *
-     * @param SqsClient        $sqsClient
-     * @param SqsQueueOptions  $options
-     * @param string           $name
+     * @param SqsClient $sqsClient
```
Comment (roelvanduijnhoven) [inline src/Queue/SqsQueue.php:32]: You can simply remove the PHPDoc if the arguments in the function are allready typed :scissors:.

**Review by roelvanduijnhoven — COMMENTED** [review]
Nice work! Had a busy week, but I'll happily want to help now to get this merged and released :).

* A library should not commit its own lock file. So you can remove the composer.lock :).
* Could you add the newer PHP versions in the TravisCI test suite? I placed a comment, showing how you can do this!

After you fixed some things, I'll merge, and test the library myself. If it passes: we'll do a release!

Excellent work :).

**Review by habibillah — COMMENTED** [review]
update dependency libraries

**Comment by roelvanduijnhoven** [comment]
Please let me know if you want me to review again @habibillah.

**Comment by habibillah** [comment]
yes please

**Comment by roelvanduijnhoven** [comment]
Allright, looks fine to me! Great work @habibillah :clap:.

My plan is to first release a beta version. You can than test and use that version yourself. If no bugs occur, we can release a new major version.

But.. before I do that. Are there any deprecations or other notable problems with this new update that need to go into the changelog?

**Comment by habibillah** [comment]
I have tested on my end, and it work well.

**Comment by habibillah** [comment]
I think you will get issue with `aws/aws-sdk-php-zf2`. that lib moved to use Laminas in v4.3. if you use composer v2 you will get conflict on `laminas/laminas-dependency-plugin`

````
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - laminas/laminas-dependency-plugin[1.0.0, ..., 1.0.4] require composer-plugin-api ^1.1 -> found composer-plugin-api[2.0.0] but it does not match the constraint.
    - aws/aws-sdk-php-zf2 4.3.0 requires laminas/laminas-dependency-plugin ^1.0 -> satisfiable by laminas/laminas-dependency-plugin[1.0.0, ..., 1.0.4].
    - Root composer.json requires aws/aws-sdk-php-zf2 ^4.3 -> satisfiable by aws/aws-sdk-php-zf2[4.3.0].
````

Fortunately, my project don't use `aws/aws-sdk-php-zf2`, but directly use `aws/aws-sdk-php`. So nothing issue in my end

**Comment by roelvanduijnhoven** [comment]
@habibillah So what you are telling me is that your PR does not use aws/aws-sdk-php-zf2? If so: let's change the dependency from `aws/aws-sdk-php-zf2` rto `aws/aws-sdk-php`. Right?

**Comment by habibillah** [comment]
No no. I mean, because of a few issues my app does not utilize `aws/aws-sdk-php-zf2`. because I save my `aws` credential in database, I simply build my factory file look like below (doesn't follow your doc)

````
class SqsClientFactory implements FactoryInterface
{
    public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
    {
        /** @var OptionsTable $optionsTable */
        $optionsTable = $container->get(OptionsTable::class);

        return new SqsClient([
            'version'     => 'latest',
            'region'      => 'us-east-1',
            'credentials' => [
                'key' => $optionsTable->fetchByName('amazon_key')['value'] ?? '',
                'secret' => $optionsTable->fetchByName('amazon_secret')['value'] ?? '',
            ],
        ]);
    }
}
````

Some people may still use `aws/aws-sdk-php-zf2` to automatically build their dependency to `aws/aws-sdk-php`, so I think it better to keep it there. I also tested using in simple app using `aws/aws-sdk-php-zf2` and it work well as long you use `composer` v1. my notes above just in case you or other user use composer v2, you may get those error.

**Comment by roelvanduijnhoven** [comment]
Allright, let's figure that out in another PR or issue. So as far as you are concerned, we can release this repo @habibillah :)?

**Comment by habibillah** [comment]
yes sure

**Comment by roelvanduijnhoven** [comment]
@habibillah I released v2.0.0-beta, so no public release, but we can test with it. Can you verify that it works for you? Let me know: if so, I will release new version soon.

**Comment by habibillah** [comment]
tested. work well on my side

**Pull request merged**